### PR TITLE
Adhereing to tenancy access roles in observability CRUD operations

### PR DESCRIPTION
**Pull request by vamsi-amazon**
Signed-off-by: vamsi-amazon <reddyvam@amazon.com>

### Description
Adhering to tenancy access roles in Observability CRUD operations.

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: opensearch-observability/src/main/kotlin/org/opensearch/observability/security/UserAccessManager.kt [hunk 1]
```diff
@@ -182,4 +202,11 @@ internal object UserAccessManager {
     private fun isUserPrivateTenant(user: User?): Boolean {
         return getUserTenant(user) == PRIVATE_TENANT
     }
+
+    private fun getUserTenantAccess(user: User?): String {
```
Comment (derek-ho) [inline opensearch-observability/src/main/kotlin/org/opensearch/observability/security/UserAccessManager.kt:206]: user should not be nullable at this point?

File: opensearch-observability/src/main/kotlin/org/opensearch/observability/action/ObservabilityActions.kt [hunk 2]
```diff
@@ -30,6 +31,12 @@ internal object ObservabilityActions {
     fun create(request: CreateObservabilityObjectRequest, user: User?): CreateObservabilityObjectResponse {
         log.info("$LOG_PREFIX:ObservabilityObject-create")
         UserAccessManager.validateUser(user)
+        if (!UserAccessManager.doesUserHasAccessToCreate(user)) {
+            throw OpenSearchStatusException(
+                "Permission denied for ObservabilityObject creation",
```
Comment (derek-ho) [inline opensearch-observability/src/main/kotlin/org/opensearch/observability/action/ObservabilityActions.kt:36]: I noticed in the exceptions below you are leaving it really generic i.e. "Permission denied for ObservailibtyObject ${id}", should this be the same?

File: opensearch-observability/src/main/kotlin/org/opensearch/observability/security/UserAccessManager.kt [hunk 3]
```diff
@@ -23,6 +25,9 @@ internal object UserAccessManager {
     private const val OPENSEARCH_DASHBOARDS_SERVER_USER = "opensearchdashboardsserver" // TODO: Change it to background user when created.
     private const val PRIVATE_TENANT = "__user__"
     const val DEFAULT_TENANT = ""
+    private const val WRITE_ACCESS = "WRITE"
```
Comment (YANG-DB) [inline opensearch-observability/src/main/kotlin/org/opensearch/observability/security/UserAccessManager.kt:28]: should this also be an enum ?

File: opensearch-observability/src/main/kotlin/org/opensearch/observability/security/UserAccessManager.kt [hunk 4]
```diff
@@ -161,6 +165,22 @@ internal object UserAccessManager {
         }
     }
 
+    /**
+     * validate if user has access to create
+     */
+    fun doesUserHasAccessToCreate(user: User?): Boolean {
+        if (user == null) { // Security is disabled
```
Comment (YANG-DB) [inline opensearch-observability/src/main/kotlin/org/opensearch/observability/security/UserAccessManager.kt:172]: isnt this a security risk ?

**Review by YANG-DB — CHANGES_REQUESTED** [review]

**Comment by vamsi-amazon** [comment]
Closing this PR. will raise once we decided to go ahead with release.

**Pull request closed without merging**